src/common/buffer.cc

```diff
@@ -1826,7 +1826,7 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     if (!_buffers.empty()) {
       ptr &l = _buffers.back();
       if (l.get_raw() == bp.get_raw() &&
-	  l.end() == bp.start() + off) {
+	  l.end() == off) {
```

Comment:
Thinks， but, I don't think so, the off here equal to append_buffer.end() - gap(the length of data that is append), so it is also the offset from the raw data.